src/m0AutomaticRoute.ts

```diff
@@ -386,6 +457,112 @@ export class M0AutomaticRoute<N extends Network>
     );
   }
 
+  async *transferSolanaExtension<N extends Network, C extends SolanaChains>(
+    ntt: SolanaNtt<N, C>,
+    sender: AccountAddress<C>,
+    amount: bigint,
+    recipient: ChainAddress,
+    sourceToken: string,
+    destinationToken: string,
+    options: Ntt.TransferOptions,
+    outboxItem?: Keypair
+  ): AsyncGenerator<SolanaUnsignedTransaction<N, C>> {
+    const router = new SolanaRoutes(ntt);
+
+    // Bridging from $M
+    if (router.getSolanaContracts().token === sourceToken) {
+      return ntt.transfer(sender, amount, recipient, options);
```

Comment:
```suggestion
      yield* ntt.transfer(sender, amount, recipient, options);
```

Doing a `return` here will set the returned generator to be empty. [`yield*`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/yield*) is what you actually want.